sdb/commands/zfs/histograms.py

```diff
@@ -0,0 +1,149 @@
+#
+# Copyright 2020 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+import argparse
+from typing import Iterable
+
+import drgn
+import sdb
+from sdb.commands.internal import fmt
+
+
+class ZFSHistogram(sdb.Command):
+    """
+    Print ZFS Histogram and print its median segment size.
+
+    NOTE
+        The median is just an approximation as we can't tell the
+        exact size of each bucket within a histogram bucket.
+
+    EXAMPLES
+        Dump the histogram of the normal metaslab class of the rpool:
+
+            sdb> spa rpool | member spa_normal_class.mc_histogram | zhist
+            seg-size   count
+            --------   -----
+              512.0B:   4359 *******************
+               1.0KB:   3328 ***************
+               2.0KB:   3800 *****************
+               4.0KB:   3536 ***************
+               8.0KB:   3983 *****************
+              16.0KB:   4876 *********************
+              32.0KB:   9138 ****************************************
+              64.0KB:   4508 ********************
+             128.0KB:   2783 ************
+             256.0KB:   1952 *********
+             512.0KB:   1218 *****
+               1.0MB:    675 ***
+               2.0MB:    486 **
+               4.0MB:    267 *
+               8.0MB:    110
+              16.0MB:     50
+              32.0MB:     18
+              64.0MB:      8
+             128.0MB:     11
+             256.0MB:    102
+            Histogram Median: 339.7MB
```

Comment:
Since this value is an approximation, perhaps we should state that when it's displayed? e.g. `Approx. Histogram Median:` or `Approx. Median:` (since "histogram" can probably be implied at this point)..

Reply:
Will change to `Approx. Median:`